schemas/static/embedded_sensornet.json

```diff
@@ -71,6 +71,35 @@
                     }
                 }
             }
+        },
+        "data_format": {
+            "description": "Structure of the measurement data gathered by the system",
+            "type": "object",
+            "properties": {
+                "tag_id": {
+                    "description": "ID of the system embedded in the drum",
+                    "type": "number"
+                },
+                "utc_time": {
+                    "description": "Starting time of the measurement",
+                    "type": "number"
+                },
+                "energy_level": {
+                    "description": "Energy storage voltage at the beginning of the message uplink",
+                    "type": "number"
+                },
+                "data": {
```

Comment:
But in this case a schema is created every time the `Energy storage voltage at the beginning of the message uplink` changes. Do I understand this correctly? 

Regarding Azure this is still not clear. But would be great to have some king of prototype system/workflow that can store data and metadata from sensors with schemas to validate them. Then this can be sampled by the "digital twin" preform  model optimization and predictions. 

If now in your work with the sensors you will follow the structure and schemas we discuss here then we can make sure that the DT will be able to use / understand your data since these schemas and metadata will be used to interpret the data you produce.

Reply:
As for the Azure integration, I completely agree that this is not clear. But I think it would be nice to have a working solution here, and then give that as an example to the azure guys. For that, we still would have to think how this solution looks like and how this integrates with the DT. In particular @gdmiron , how do we think this interaction should work? We started one repo, where I integrated a pipeline as a starting point for this joint work. Then another repo (this one) was created that should cover the data schema. From my personal point of view, there is currently only very few examples (both models and data) that I think it is really additional overhead to work with two separate repos. In addition, it might really make sense to have a meeting (not sure who should participate) to outline the idea of that repo(s) and distribute the work - IMO this would also require a regular meeting at least on a biweekly schedule in order to get something done. 
Coming back to the above issue, maybe the structure of the data we should/would send should be discussed in such a meeting, I think it makes pro and cons much more obvious than in an online discussion. Maybe a meeting after April 17th?

Reply:
For me a meeting after April 17th is fine, maybe you can schedule a date in the next Friday's 7.5 meeting (which I cannot attend). 


@gdmiron, not really. The idea is to have just one schema shared by two implementations. 

* `metadata.json`: the schema describing the overall system;
* `static.json`: implementation of the static objects, in the `embedded_sensornet` example they could be the objects `drum_id` and `components`;
* `dynamic.json`: implementation of the object `data_format`, i.e. the timeseries.

e.g. If I perform two measurements, I will have two packets of data, each one with the structure `data_format` with values like `energy_level` changing. It's possible to validate the received data packet with the object `data_format` found in the metadata schema.

Reply:
I'm still a little puzzled with the static metadata. I thought static is the metadata, but apparently there is another metadata.json object planned. The schemas and examples files now include facility, drum, sensornet which are currently separated definitions that are not linked. What is the plan here? I thought we build a single json file that might include all the meta information required, and another file (e.g. a csv) that just has the measured values (and time stamp) as columns with a header that allows to determine the corresponding sensor (in the metadata.json).

Reply:
Should we have the meeting together with Suresh on the ASR model and their data integration? I feel this should be a larger topic for 7.x and thus being coordinated on that level. @gdmiron could you ask Rainer Daehn on howto proceed?

Reply:
Nothing planned yet, I'm just proposing how we could architecture the data handling.
Do you mean to have a single json file including all the sensor-systems information (embedded_sensornet, gamma detectors etc) and then csv with the values? I honestly don't agree with this implementation.
I think it would be cleaner and easier to maintain to keep the sensor-systems files separated. In this scenario each system will manifest itself with a json schema describing all its info (`metadata.json` in the previous answer, or  `embedded_sensornet.json` in the schemas/static folder in my case) and then we'll have an implementation for each time the sensor is installed (static info), or the system upload a measurement (dynamic info / timeseries / data packet).

The csv format has some limitations, that's why I would include the description of the timeseries in the metadata file (`data_format `object), it can also be a plain object just describing the items of the csv's header, if some system uploads a csv file.
Anyway, the Azure IoT API already used by Andrea and soon by me to upload the measurement timeseries, requires a json object for the upload, so a schema for the validation would be required.
That's why I'm proposing this architecture, it's well-connected with the upload implementation and generic enough to be used by many systems.

Reply:
The metadata for the sensors has to be placed into a joint database, which means that in the database we do not have multiple files, but a joint structure that would allow to do a query e.g. "Please give me all the sensors for drum X and location Y that measure temperature and mechanical strain. Also, return the position and the orientation of the sensors and the coordinate system". So if we put all sensors into a single file, or upload them separately and then aggregate them afterwards (which would be my preferred solution), the schemas for both metadata are still identical.
But I agree, we should try to merge as soon as possible and then move this discussion somewhere to an issue maybe. In order to merge, all the comments have to be resolved first.

Reply:
I still think that a single file is not a good solution, it's better to upload a file for each system and then perform an aggregation (by a script) or, if the systems have a similar structure (at least a json object called `sensors`), an iteration over all the available files and look for the `sensors` object.
Let's try to clarify this topic in the next meeting. Has it already been scheduled?

Reply:
just got an invitation for May 2nd at 9:30am.

Reply:
As for the response above, I don't think that we would search in files anyway. The structure that we are developing is just for uploading the data to the database, everything should be aggregated afterwards. Thus, the structure (of the json file) should not be used for retrieving the data at all. In addition, IMO, the structure of a json file should be identical independent of the number of sensor information that is uploaded, since it just includes a graph, and whether this graph is created before or after uploading should not make a difference.

Reply:
Well, maybe I don't have a clear picture of the whole pipeline. Anyway, let's keep the timeseries out of the metadata file for the moment and try to merge the open pull requests (this it's still blocked for the changes requests).
Then let's try to have a meeting a clarify as much as possible.